hypervisor/esxi/esxi.go

```diff
@@ -0,0 +1,309 @@
+	"github.com/axsh/openvdc/hypervisor"
+	"github.com/axsh/openvdc/model"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	cli "github.com/vmware/govmomi/govc/cli"
+	_ "github.com/vmware/govmomi/govc/datastore"
+	_ "github.com/vmware/govmomi/govc/vm"
+	_ "github.com/vmware/govmomi/govc/vm/guest"
+	_ "github.com/vmware/govmomi/govc/device"
+	_ "github.com/vmware/govmomi/govc/device/serial"
+	"golang.org/x/crypto/ssh"
+)
+
+type BridgeType int
+
+const (
+	None BridgeType = iota // 0
+	Linux
+	OVS
+)
+
+var settings struct {
+	ScriptPath      string
+	EsxiUser        string
+	EsxiPass        string
+	EsxiIp          string
+	EsxiDatacenter  string
+	EsxiInsecure    bool
+	EsxiHostSshkey  string
+	EsxiVmUser      string
+	EsxiVmPass      string
+	EsxiVmDatastore string
+	EsxiUrl         string
+	BridgeName      string
+	BridgeType      BridgeType
+}
+
+func (t BridgeType) String() string {
+	switch t {
+	case Linux:
+		return "linux"
+	case OVS:
+		return "ovs"
+	default:
+		return "none"
+	}
+}
+
+type EsxiMachine struct {
+	SerialConsolePort int
+}
+
+type EsxiHypervisorProvider struct {
+}
+
+type EsxiHypervisorDriver struct {
+	hypervisor.Base
+	template  *model.EsxiTemplate
+	machine   *EsxiMachine
+	imageName string
+	hostName  string
+	vmName    string
+}
+
+func (p *EsxiHypervisorProvider) Name() string {
+	return "esxi"
+}
+
+func init() {
+	hypervisor.RegisterProvider("esxi", &EsxiHypervisorProvider{})
+	viper.SetDefault("hypervisor.esxi-insecure", true)
+}
+
+func (p *EsxiHypervisorProvider) LoadConfig(sub *viper.Viper) error {
+	if sub.IsSet("bridges.name") {
+		settings.BridgeName = sub.GetString("bridges.name")
+		if sub.IsSet("bridges.type") {
+			switch sub.GetString("bridges.type") {
+			case "linux":
+				settings.BridgeType = Linux
+			case "ovs":
+				settings.BridgeType = OVS
+			default:
+				return errors.Errorf("Unknown bridges.type value: %s", sub.GetString("bridges.type"))
+			}
+		}
+	} else if sub.IsSet("bridges.linux.name") {
+		log.Warn("bridges.linux.name is obsolete option")
+		settings.BridgeName = sub.GetString("bridges.linux.name")
+		settings.BridgeType = Linux
+	} else if sub.IsSet("bridges.ovs.name") {
+		log.Warn("bridges.ovs.name is obsolete option")
+		settings.BridgeName = sub.GetString("bridges.ovs.name")
+		settings.BridgeType = OVS
+	}
+
+	settings.ScriptPath = sub.GetString("hypervisor.script-path")
+	settings.EsxiUser = sub.GetString("hypervisor.esxi-user")
+	settings.EsxiPass = sub.GetString("hypervisor.esxi-pass")
+	settings.EsxiIp = sub.GetString("hypervisor.esxi-ip")
+	settings.EsxiDatacenter = sub.GetString("hypervisor.esxi-datacenter")
+	settings.EsxiInsecure = sub.GetBool("hypervisor.esxi-insecure")
+	settings.EsxiHostSshkey = sub.GetString("hypervisor.esxi-host-sshkey")
+	settings.EsxiVmUser = sub.GetString("hypervisor.esxi-vm-user")
+	settings.EsxiVmPass = sub.GetString("hypervisor.esxi-vm-pass")
+	settings.EsxiVmDatastore = sub.GetString("hypervisor.esxi-vm-datastore")
```

Comment:
Should validate the required parameters and then return error if they were blank.